executor/executor_test.go

```diff
@@ -5441,41 +5443,42 @@ func TestHistoryReadInTxn(t *testing.T) {
 					}
 
 					for _, sql := range notAllowedSQLs {
-						err := tk.ExecToErr(sql)
+						err := tkt.ExecToErr(sql)
 						require.Errorf(t, err, "can not execute write statement when 'tidb_snapshot' is set")
+						time.Sleep(20 * time.Millisecond)
```

Comment:
The real problem of this data race, is that the query result set is not closed.
The leaked coprocessor goroutine still working in the background, and then race with the next SQL statement on `session.SessionVars`

It's fixed accidently by this PR here https://github.com/pingcap/tidb/pull/34837/files#diff-cda21674d674295dde958bfdfe5d6ecd21ae07fd310fa907920e1214108d974fR596


Sleep 20ms is not the correct workaround, it just reduce the change of DATA RACE